Bug: /api/analytics/codebase/sources/summary returns 404 — FastAPI route ordering collision

## Problem

`GET /api/analytics/codebase/sources/summary` returns HTTP 404. The Codebase Analytics landing page fails to load source summaries.

```
[ERROR] [errorHandler] [HTTP 404] Resource not found: /api/analytics/codebase/sources/summary
```

## Root Cause

**FastAPI route ordering bug** in `sources.py`. Routes are matched in registration order:

- **Line 258**: `@router.get("/sources/{source_id}")` — registered FIRST
- **Line 316**: `@router.get("/sources/summary")` — registered SECOND (never reached)

When `/sources/summary` is requested, FastAPI matches `{source_id}` with value `"summary"`. The `get_code_source("summary")` handler finds no source with that ID → 404.

The docstring on line 327 even says *"Registered before /sources/{source_id}/summary to avoid the literal string 'summary' being captured"* — but it's only before the per-source summary route (`/sources/{source_id}/summary`), not the catch-all `{source_id}` route above it.

## Fix

Move the `get_all_source_summaries()` function (lines 316-332) above `get_code_source()` (line 258). The correct route registration order should be:

```
@router.get("/sources")              # line 210 — list all
@router.post("/sources")             # line 220 — create
@router.get("/sources/summary")      # MOVED HERE — batch summary (literal path)
@router.get("/sources/{source_id}")  # line 258 — get by ID (parameterized)
@router.put("/sources/{source_id}")  # line 272 — update
@router.delete("/sources/{source_id}") # line 295 — delete
...
```

## Files

- `autobot-backend/api/codebase_analytics/endpoints/sources.py` (lines 258, 316-332)
- `autobot-frontend/src/components/analytics/CodebaseAnalyticsLanding.vue` (line 247 — frontend caller)

## Impact

Codebase Analytics landing page cannot load source summaries. Falls back to N+1 per-source calls or shows empty state.

--- a/autobot-backend/api/codebase_analytics/endpoints/sources.py
+++ b/autobot-backend/api/codebase_analytics/endpoints/sources.py
@@ -255,6 +255,25 @@ async def create_code_source(request: CodeSourceCreateRequest):
     return JSONResponse(source.model_dump(), status_code=201)
 
 
+@router.get("/sources/summary")
+@with_error_handling(
+    category=ErrorCategory.SERVER_ERROR,
+    operation="batch_source_summary",
+    error_code_prefix="CODEBASE",
+)
+async def get_all_source_summaries():
+    """Return summaries for all sources in one request (#1468).
+
+    Replaces N+1 per-source /summary calls from the landing page.
+    Registered before /sources/{source_id} to avoid the literal
+    string "summary" being captured by the {source_id} path parameter (#2654, #3107).
+    """
+    sources = await list_sources()
+    results = await asyncio.gather(*[_build_summary(s) for s in sources])
+    summaries = {r["source_id"]: r for r in results}
+    return JSONResponse({"summaries": summaries})
+
+
 @router.get("/sources/{source_id}")
 @with_error_handling(
     category=ErrorCategory.SERVER_ERROR,
@@ -311,25 +330,6 @@ async def delete_code_source(source_id: str):
     ok = await delete_source(source_id)
     logger.info("Deleted code source %s", source_id)
     return JSONResponse({"success": ok, "source_id": source_id})
-
-
-@router.get("/sources/summary")
-@with_error_handling(
-    category=ErrorCategory.SERVER_ERROR,
-    operation="batch_source_summary",
-    error_code_prefix="CODEBASE",
-)
-async def get_all_source_summaries():
-    """Return summaries for all sources in one request (#1468).
-
-    Replaces N+1 per-source /summary calls from the landing page.
-    Registered before /sources/{source_id}/summary to avoid the literal
-    string "summary" being captured by the {source_id} path parameter (#2654).
-    """
-    sources = await list_sources()
-    results = await asyncio.gather(*[_build_summary(s) for s in sources])
-    summaries = {r["source_id"]: r for r in results}
-    return JSONResponse({"summaries": summaries})
 
 
 @router.post("/sources/{source_id}/sync")

